### DISCUSSION TITLE
How can i chain selectors in async mode?

### DISCUSSION THREAD

**Discussion by tschuliiia** [webdriverio/webdriverio#7110] [Q&A]
i am trying to access a nested element in an angular application which can't be identified uniquely within a container that can be identified uniquely. so in concrete, i want to click the button within the app-team component. I have multiple app-team components, which can be identified uniquely but with the same button.

my structure is the following:
```
<app-team _ngcontent-gxy-c1="" data-test-id="team_container" _nghost-gxy-c2="" ng-reflect-fights="" ng-reflect-team-index="0">
<form>
<button _ngcontent-gxy-c2="" class="primary"> Create a team </button>
</form>
</app-team>
```
my first idea was either chaining the selectors, such as the following, but it doesn't compile :
```
const team1 = await $('app-team[data-test-id="team-container"][ng-reflect-team-index="0"]');
await team1.$('button').click
```

second idea:
```
    const team1 = await $('app-team[data-test-id="team-container"][ng-reflect-team-index="0"]');
    const button = await team1.selectByAttribute('class','primary' );
    await button.click();
```

unfortunately i can't find any examples for how to do selector chaining within async mode, only for sync mode. 
Does anybody has an idea how to solve the issue?

Thanks,
julia

**Comment by christian-bromann** [comment]
Hey Julia 👋 ,

> my first idea was either chaining the selectors, such as the following, but it doesn't compile :

You can technically chain this way, do you have an error message of the compile error?

> unfortunately i can't find any examples for how to do selector chaining within async mode, only for sync mode.

As we currently transition off sync mode we work towards creating the same API. Right now the difference with async is that you will need to do this:

```js
await (await (await $("foo")).$$('bar'))[0].$("loo"))
```

Your examples above look fine. It just becomes annoying if you chain them in one line. Again, we are about to fix this soon.

Do you have any further questions?
